contracts/tips/src/contract.rs

```diff
@@ -0,0 +1,1728 @@
+        ))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut<DesmosQuery>,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    msg.validate()?;
+
+    match msg {
+        ExecuteMsg::SendTip { target } => execute_send_tip(deps, info, target),
+        ExecuteMsg::UpdateServiceFee { new_fee } => execute_update_service_fee(deps, info, new_fee),
+        ExecuteMsg::UpdateAdmin { new_admin } => execute_update_admin(deps, info, new_admin),
+        ExecuteMsg::UpdateSavedTipsRecordThreshold { new_threshold } => {
+            execute_update_saved_tips_record_threshold(deps, info, new_threshold)
+        }
+        ExecuteMsg::ClaimFees { receiver } => execute_claim_fees(deps, env, info, receiver),
+    }
+}
+
+fn execute_send_tip(
+    deps: DepsMut<DesmosQuery>,
+    info: MessageInfo,
+    target: Target,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+
+    // Computes the fee and the coins to be sent to the user
+    let (_, coin_to_send) = match config.service_fee {
+        None => (Vec::<Coin>::new(), info.funds),
+        Some(service_fee) => service_fee.compute_fees(info.funds)?,
+    };
+
+    // Resolve the receiver and the optional post id
+    let (post_id, receiver) = match target {
+        Target::ContentTarget { post_id } => {
+            let post = PostsQuerier::new(deps.querier.deref())
+                .query_post(config.subspace_id, post_id.u64())?
+                .post;
+            (post.id.u64(), post.author)
+        }
+        Target::UserTarget { receiver } => {
+            // Set the post id to 0 (invalid id) to signal that this tip is referencing an user
+            (0_u64, deps.api.addr_validate(&receiver)?)
+        }
+    };
+
+    if config.tips_record_threshold > 0 {
+        let tip_key: TipsRecordKey = (info.sender.clone(), receiver.clone(), post_id);
+        let tip_record_key = TIPS_RECORD.key(tip_key.clone());
+        let tip_record_coins = if let Some(mut coins) = tip_record_key.may_load(deps.storage)? {
+            // Append the new coins
+            coins.extend(coin_to_send.clone());
+            utils::merge_coins(coins)?
+        } else {
+            // Load the key list
+            let mut tips_key_list = TIPS_KEY_LIST.load(deps.storage)?;
+            // If we have reached the threshold remove the oldest key
+            if tips_key_list.len() == config.tips_record_threshold as usize {
+                TIPS_RECORD.remove(deps.storage, tips_key_list.pop_front().unwrap());
+            }
+            // Add the new key to the end
+            tips_key_list.push_back(tip_key);
+            TIPS_KEY_LIST.save(deps.storage, &tips_key_list)?;
+            // Return the new coins
+            coin_to_send.clone()
+        };
+        tip_record_key.save(deps.storage, &tip_record_coins)?;
+    }
+
+    if coin_to_send.is_empty() {
```

Comment:
It can be moved before the line 138 `if config.tips_record_threshold > 0` to check the funds is not zero first.